### PR TITLE
SJSU-1067_2 check booking for not logged in user 

### PR DESCRIPTION
**Pull request by minhphan156**
What I did:
- added new page BookingNotLoggedIn
- allow user to access private history route when authenticated with bookingID and lastName
- changed backend bookin.js to work with current historyOverView

How to test:
- use **5cc80f24e85a88435461775a** as booking id and **Stark** as last name to check booking

### CODE REVIEW

File: front_end/src/actions/profileActions.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ import {
   PROFILE_LOADING,
   CLEAR_CURRENT_PROFILE,
   SET_CURRENT_USER,
-  GET_HISTORY
+  GET_HISTORY,
+  BOOKING_NOT_LOGGED_IN_AUTHENTICATED
```
Comment (SandroSal) [inline front_end/src/actions/profileActions.js]: this sounds confusing... how is he not logged in authenticated?

Comment (SandroSal) [reply front_end/src/actions/profileActions.js]: oh, is this that you checked the ID and name?

File: front_end/src/components/auth/BookingNotLoggedIn.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import { connect } from "react-redux";
+import Grid from "@material-ui/core/Grid";
+
+import { getHistoryNotLoggedIn } from "../../actions/profileActions";
+import TextFieldGroup from "../common/TextFieldGroup";
+import Paper from "@material-ui/core/Paper";
+import AbodeLogo from "../../images/logo.png";
+import "./auth.css";
+
+class BookingNotLoggedIn extends Component {
```
Comment (SandroSal) [inline front_end/src/components/auth/BookingNotLoggedIn.js]: don't know if its good to use the same widget as for login...

File: front_end/src/components/history/HistoryOverview.js [hunk 3]
```diff
@@ -83,8 +84,11 @@ const styles = {
 
 class HistoryOverview extends Component {
   componentWillMount() {
-    this.props.getHistory();
-    this.props.getProfileInfo();
+    if (this.props.auth.isAuthenticated) {
+      // only call these if user logged in
+      this.props.getHistory();
```
Comment (SandroSal) [inline front_end/src/components/history/HistoryOverview.js]: if you reuse this file, it has to be updated... it shouldn't show rewardspoints used or rewardspoints earned.

File: front_end/src/components/layout/Navbar.js [hunk 4]
```diff
@@ -52,6 +52,13 @@ class Navbar extends Component {
             </Button>
           </Link>
         </Grid>
+        <Grid item>
+          <Link to="/booking-not-logged-in">
```
Comment (SandroSal) [inline front_end/src/components/layout/Navbar.js]: please change the way of displaying the button/link
I'd make it a link (similar to Deals of the Week), but show this link for all page sizes. (deal and feat. city is not displayed on smaller screens)

File: routes/api/bookin.js [hunk 5]
```diff
@@ -5,20 +5,19 @@ const passport = require("passport");
 const Hotel = require("../../models/Hotel");
 const Booking = require("../../models/booking");
 const Customer = require("../../models/customer");
-const City = require("../../models/city")
+const City = require("../../models/city");
 const User = require("../../models/User");
 
-const confirmEmail = require('../../email/confirmationEmail')
-const welcomeEmail = require('../../email/welcomeEmail')
+const confirmEmail = require("../../email/confirmationEmail");
```
Comment (SandroSal) [inline routes/api/bookin.js]: 1000 lines changed... don't feel like going through all this... what did you change?

File: routes/api/bookin.js [hunk 6]
```diff
@@ -28,500 +27,570 @@ router.post("/review", (req,res) => {
...
-
-                    }); // end map()
-                
-                }); // end Customer.findById
-
-            }); // end Booking.find
-        
-});
-
-// @route GET /api/booking/guest-history
```
Comment (minhphan156) [inline routes/api/bookin.js]: the changes started here

File: routes/api/bookin.js [hunk 7]
```diff
@@ -28,500 +27,570 @@ router.post("/review", (req,res) => {
...
-// @desc History page
-// @access public
-router.get("/guest-history", (req, res) => {
```
Comment (minhphan156) [inline routes/api/bookin.js]: the changes started here

File: routes/api/bookin.js [hunk 8]
```diff
@@ -28,500 +27,570 @@ router.post("/review", (req,res) => {
...
+        }); // end map()
+      }); // end Customer.findById
+    }); // end Booking.find
+  }
+);
 
+// @route GET /api/booking/guest-history
+// @desc History page
+// @access public
+router.post("/guest-history", (req, res) => {
```
Comment (minhphan156) [inline routes/api/bookin.js]: the changes started here

File: routes/api/bookin.js [hunk 9]
```diff
@@ -28,500 +27,570 @@ router.post("/review", (req,res) => {
...
+            rewardDiscount: book.rewardDiscount,
+            reservedDate: book.reservedDate,
+            nightlyRate: book.numOfNights,
+            starReview: book.starReview,
+            comment: book.review
+          });
+          // Give frontend the goods
+          return res.status(200).send(historyPack);
+        });
+      })
```
Comment (minhphan156) [inline routes/api/bookin.js]: the changes stopped here

File: routes/api/bookin.js (hunk 5 above)
Comment (minhphan156) [reply routes/api/bookin.js]: I made changes only for **router.post("/guest-history", (req, res) => { ...} ** . Please look down below for my comments

**Comment by SandroSal** [comment]
- links are to close together for LG and MD
- check booking doesnt show up for SM

**Comment by SandroSal** [comment]
To test:
- try different sizes on landing while logged in (should not show 'search booking')
- try different sizes on landing while not logged in (always possible to 'search booking')
- actually click on search booking (try different sizes)
- add combination of valid/non-valid confimation# and lastname
- booking-overview (try different sizes)

**Pull request merged**